### ISSUE TITLE
Integrated Gradient without edge attribute

### ISSUE THREAD

**Issue by dongwei2001** [pytorch/captum#1296]
## ❓ Questions and Help

Hi! I'm facing a problem: We want to use Integrated gradient(IG) to explain a GCN model. We want to generate edge_mask for each edge in the network. But this GCN model doesn't make use of edge attribute during both training and testing. 
So my question is: Can we use IG to explain this model without using edge attribute? Or I can only switch to other explainers?

Thanks so much!

**Comment by craymichael** [comment]
Apologies for the delay.

If the graph model has no differentiable edge feature, Integrated Gradients cannot directly attribute an "edge mask" unless you introduce one into the forward computation.

A common pattern is to add a differentiable mask that multiplies each edge's message:

```python
message = edge_mask[:, None] * message
```

Then you can attribute the model output with respect to `edge_mask`, using a baseline such as zeros or another reference mask. That gives attribution over edge presence / message strength.

If you do not want to change the model forward path, a perturbation method is usually more natural: remove / keep edges, evaluate the output change, and treat that as edge importance. IG needs a continuous input path, while edge existence in the original graph is discrete.

This should answer the question, so I'll close this for now. Please feel free to reply with any follow-up questions.
